### PR TITLE
Feature/show idea

### PR DESCRIPTION
**Pull request by reshetnev**
Review.
Has added link to deatails page from photo.

### CODE REVIEW

File: src/main/webapp/js/ideas/showIdea.directive.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		.module('app.directives')
+		.directive('showIdea', showIdea);
+
+	function showIdea() {
+		return {
+			restrict: 'E',
+			replace: true,
+			scope: {
+				idea: '@',
+				photo: '@',
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/ideas/showIdea.directive.js]: Will each idea have its own photo? Or there is the same photo for all?

File: src/main/webapp/js/ideas/showIdea.directive.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		.directive('showIdea', showIdea);
+
+	function showIdea() {
+		return {
+			restrict: 'E',
+			replace: true,
+			scope: {
+				idea: '@',
+				photo: '@',				
+				title: '@',
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/ideas/showIdea.directive.js]: If idea object is already passed to the directive, may be, other data, like title, description and other, can be obtained from idea object and there is no necessity to pass it to the directive?

File: src/main/webapp/pages/app.html [hunk 3]
```diff
@@ -6,37 +6,17 @@
...
-                        <span ng-bind-html="idea.description | highlightTextInDescriptionOfIdea:ideaCtrl.criteria"></span>
-
-                    </p>
-                    <p class ="end-text">
-                        <span class ="small-area"><span class="glyphicon glyphicon-time"></span>{{ idea.createdAt | date:'dd.MM.yy'}}</span>
-                        <span class ="small-area"><span class="glyphicon glyphicon-user"></span>{{idea.author.username}}</span>
-                        <span class ="small-area"><span class="glyphicon glyphicon-tag"></span><span ng-repeat="tag in idea.tags">{{tag.name}}</span></span></p>
-                </div>
-
+                <show-idea idea="{{idea}}" photo="{{ideasCtrl.photo}}" title="{{idea.title}}" details="ideasCtrl.details(idea)"
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/pages/app.html]: to use directive 'show-idea' user have to know too many details

File: src/main/webapp/js/ideas/showIdea.directive.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+				description: '@',
+				created: '@',
+				author: '@',			
+				tag: '@',
+				rtng: '@',
+				ngModel: '=',
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/ideas/showIdea.directive.js:17]: May be, it makes sense to make required methods belong to directive: move required methods from ideaController to directive's link function (if they are not required in other places except directive)

File: src/main/webapp/js/ideas/showIdea.directive.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+(function() {
+	'use strict';
+
+	angular
+		.module('app.directives')
+		.directive('showIdea', showIdea);
```
Comment (Hanna-Aliakseichykava) [inline src/main/webapp/js/ideas/showIdea.directive.js:6]: Please, add tests for new directive

File: src/test/javascript/e2e/scenarios.js [hunk 6]
```diff
@@ -21,5 +21,37 @@ describe('appIdea', function() {
...
+		browser.get('http://evbymogsd0030.minsk.epam.com:7080/idea');
+		//get first element by attribute ng-click="details()" (photo)
+		element.all(by.css('a[ng-click="details()"]')).first().click();
+		browser.getLocationAbsUrl().then(function(url) {
+			expect(url.split('%')[0].split('#')[1]).toBe('/ideaDetails');
+		});
+	});
+	
+	it('should redirect  to ideaDetails page', function() {
+		browser.get('http://evbymogsd0030.minsk.epam.com:7080/idea');
```
Comment (Hanna-Aliakseichykava) [inline src/test/javascript/e2e/scenarios.js]: May be it is possible to use constant variable for url

**Comment by Hanna-Aliakseichykava** [comment]
:+1:

File: src/test/javascript/e2e/scenarios.js (hunk 6 above)
Comment (reshetnev) [inline src/test/javascript/e2e/scenarios.js]: I agree.
I'm correcting now.

**Comment by alxx88** [comment]
:+1:

**Comment by LostFan** [comment]
:+1:

**Pull request merged**